### PR TITLE
✨ Implement backend

### PR DESCRIPTION
**Pull request by phlgr**

### CODE REVIEW

**Review by Seborider — APPROVED** [review]
🥇

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
   "dependencies": {
     "dotenv": "^10.0.0",
     "express": "^4.17.1",
+    "node-fetch": "^2.6.1",
```
Comment (Seborider) [inline package.json:50]: What does node-fetch do?

File: src/lib/genreMap.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export const GENRES = {
```
Comment (Seborider) [inline src/lib/genreMap.ts:1]: What is the lib directory for?

**Comment by BenFielmann** [comment]
Mighty skills, I also want to reach this skill level 😅

File: src/lib/models/movie.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    id: movie.id,
+    genres: movie.genres?.map((genre) => genre.name),
+  };
+}
+
+export async function searchMovie(
+  query: string,
+  genre?: string
+): Promise<Movie[]> {
+  const genreId = (Object.keys(GENRES) as GenreID[]).find(
```
Comment (Seborider) [inline src/lib/models/movie.ts:77]: What does `as` do?

**Review by aronvaupel — COMMENTED** [review]
There are two files named movie.ts. Is that by purpose?

**Review by Seborider — APPROVED** [review]
💯

File: src/lib/models/movie.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+export async function getMovie(id: string): Promise<Movie> {
+  const response = await fetch(
+    `https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`
+  );
+
+  const movie: MovieDetailsFromAPI = await response.json();
+
+  if (!movie.title) {
```
Comment (aronvaupel) [inline src/lib/models/movie.ts:59]: And what will a "normal" user, who does not open dev tools, see?

File: src/lib/models/movie.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    return {
+      title: movie.title,
+      rating: movie.vote_average,
+      overview: movie.overview,
+      image: `https://image.tmdb.org/t/p/w500${movie.poster_path}`,
+      id: movie.id,
+      genres: movie.genre_ids.map((id: GenreID) =>
+        GENRES[id] ? GENRES[id] : ''
+      ),
+    };
```
Comment (joshua-meyerrr) [inline src/lib/models/movie.ts:27]: You reapeat this snippet multiple times. Maybe consider creating a formatMovies function.

**Review by joshua-meyerrr — COMMENTED** [review]
Well done 🚀

File: src/lib/routes/movie.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    const popularMovies = await getPopularMovies();
+    res.json(popularMovies);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+router.get('/latest', async (_req, res) => {
+  const latestMovies = await getLatestMovies();
+  res.json(latestMovies);
```
Comment (danifosy) [inline src/lib/routes/movie.ts:23]: why don't we use try/catch here?

**Review by marwinburesch — APPROVED** [review]
Is OK

**Pull request merged**